crates/wascc-host/src/actors/actor_host.rs

```diff
@@ -47,15 +61,59 @@ impl ActorHost {
         });
 
         match guest {
-            Ok(g) => ActorHost {
-                guest_module: g,
-                claims: c,
-                mw_chain,
-                image_ref,
-            },
+            Ok(g) => {
+                let c = c3.clone();
+                let entity = WasccEntity::Actor(c.subject.to_string());
+                let b = MessageBus::from_registry();
+                let b2 = b.clone();
+                let recipient = ctx.address().clone().recipient();
+                let _ = block_on(async move {
+                    b.send(Subscribe {
+                        interest: entity,
+                        subscriber: recipient,
+                    })
+                    .await
+                });
+                let pc = PutClaims{ claims: c.clone() };
+                let r = block_on(async move {
+                    if let Err(e) = b2.send(pc).await {
+                        error!("Actor failed to advertise claims to bus: {}", e);
+                        ctx.stop();
+                        Err("Failed to advertise bus claims".into())
```

Comment:
So the only situation where an actor _should_ fail to start is if it fails to advertise its claims, e.g. capability contract IDs that it's allowed to interact with? What if we started it with no claims in the event that it fails to advertise? I'm specifically referring to your comment that an actor should be able to start 100% of the time even with invalid state

Reply:
An actor host now has 3 states - "just started (no state)", "running (state)", and "stopping/stopped"
there's no downside to the microscopic gap between just started and running with state, because if converting to running with state fails, the actor host stops itself.

Reply:
there's a semantic difference here that's likely confusing. The `ActorHost` _actix actor_ must never fail to start. An actor _can_ fail to initialize its state and take care of all the prep (subscribng to the bus, creating an actor object, etc)

Reply:
The semantics are getting to me a bit. Our previous validation of waSCC actors is not affected by the Actix sentiment, and it's totally fine for us to deny a waSCC actor from starting, but undesirable for the Actix actor (ActorHost) to fail to start.

Reply:
Just confirming what I'm understanding now... maybe I can call them wactors / wascctors and aactors / actixcors to keep them straight in my head

Reply:
That's correct. In Actix, you should consider it a _requirement_ that an actor be able to "start" because that really just requires you supplying an instance to a factory. In other words, the factory cannot fail to produce an "empty" actor. Once the "empty" actor has been cut from the factory, it can then have any state you like, which can include "failed to initialize"

Reply:
I struggled as well, the `*Host` structs are all Actix actors that wrap wascc abstractions like actors and providers.

Reply:
And yes, our previous validation should still exist - you should be able to find the calls to the authorization checker once we establish an instance of an actor and its claims.

Reply:
Great, thanks for the explanation 😄